src/main/jbake/content/blog/2017/07/opentracing-ejb.adoc

```diff
@@ -0,0 +1,397 @@
+to automatically trace EJB invocations.
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-07-28-teaser.png[Jaeger with EJB traces]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-07-28-teaser.png[Jaeger with EJB traces]
+endif::[]
+
+For this demo, we'll generate a project using the Wildfly Swarm project generator, which allows us to
+have a seed project with the appropriate OpenTracing support in place. The concrete OpenTracing solution
+we will use is provided by the link:https://uber.github.io/jaeger/[Jaeger project], which is also
+provided as a Wildfly Swarm Fraction.
+
+With that, we'll create a simple JAX-RS endpoint with an EJB facet, invoking a set of EJB services in
+different ways to demonstrate all the features of this integration.
+
+Our application has one endpoint called `/order`, responsible for receiving requests to place orders
+in our system. When this endpoint is called, we call a few EJB services, like:
+
+* `AccountService` to send a notification that an order has been placed
+* `OrderService` to effectively place the order
+* `InventoryService` to change our inventory
+* `InventoryNotificationService`, designed to notify other backend systems.
+
+As we are only interested in the tracing parts, we'll not implement the business code itself, only the
+scaffolding.
+
+Our demo project is heavily based on the 
+link:https://github.com/opentracing-contrib/java-ejb/tree/release-0.0.2/opentracing-ejb-example[`opentracing-ejb-example`]
+from the repository link:https://github.com/opentracing-contrib/java-ejb[`opentracing-contrib/java-ejb`].
+
+== The seed project
+
+To generate the seed project, open the link:http://wildfly-swarm.io/generator/[Wildfly Swarm generator]
+and create a project with the "Group ID" `io.opentracing.contrib.ejb` and "Artifact ID"
+`demo-example`. Add the dependencies `EJB`, `CDI`, `JAX-RS`, `OpenTracing`, and `Jaeger`.
+
+Click on `Generate Project` and you'll get a ZIP file with the seed project. Uncompress it and add the following
+dependency to the `pom.xml`, within the `dependencies` node and after the `WildFly Swarm Fractions` dependencies:
+
+```xml
+    <dependency>
+      <groupId>io.opentracing.contrib</groupId>
+      <artifactId>opentracing-ejb</artifactId>
+      <version>0.0.2</version>
+    </dependency>
+```
+
+It's now a good time to perform a sanity build, to make sure everything is in place. The first build might take a few minutes:
+
+```bash
+$ mvn wildfly-swarm:run
+...
+...
+2017-07-25 15:54:08,036 INFO  [org.wildfly.swarm] (main) WFSWARM99999: WildFly Swarm is Ready
+```
+
+If it looks good, stop the server with `Ctrl+C` and let's start coding our application!
+
+== The application
+
+Let's start by defining our JAX-RS application. This is a regular JAX-RS `Application` and we have nothing very special here:
+
+`src/main/java/io/opentracing/contrib/ejb/example/EjbExampleApplication.java`:
```

Comment:
When I generate the project using the Wildfly-Swarm generator following the instructions above it uses the package "io/opentracing/contrib/ejb/demoexample/rest"